public/app/plugins/panel/piechart/PieChartPanel.tsx

```diff
@@ -93,14 +93,20 @@ function getLegend(props: Props, displayValues: FieldDisplay[]) {
         return b.display.numeric - a.display.numeric;
       }
     })
-    .map<VizLegendItem>((value, idx) => {
-      const hidden = value.field.custom.hideFrom.viz;
+    .map<VizLegendItem | undefined>((value, idx) => {
+      const hideFromViz = value.field.custom.hideFrom.viz;
+      const hideFromLegend = value.field.custom?.hideFrom?.legend;
+
+      if (hideFromLegend) {
+        return undefined;
+      }
+
```

Comment:
Maybe
```suggestion
      const hideFrom = (value.field.custom?.hideFrom ?? {}) as HideSeriesConfig;
      if (hideFrom.legend) {
        return undefined;
      }

      const hideFromViz = Boolean(hideFrom.viz)
```
it has the hideFrom time involved, so a little nicer

Reply:
That is nicer. Couldn't commit your suggestion as it was missing the import for HideSeriesConfig - so added it manually.

Reply:
@ryantxu looks like https://github.com/grafana/grafana/pull/50229 broke the main branch. Tons of errors in betterer that are nothing to do with this PR.

Reply:
Actually ran into a betterer check for your suggestion - it is adding a type assertion:

![image](https://user-images.githubusercontent.com/434655/175490153-0d2d0cbc-d76a-40f8-9283-b6e25a19edef.png)